sig-cli/charter.md

```diff
@@ -11,7 +11,86 @@ libraries to interface with Kubernetes API's.
+Kubernetes APIs and Resources.
+
+Building tools that obfuscate the underlying Resources and APIs (e.g. through
+generalized templating or DSLs) is an anti-goal of SIG CLI.  Notable examples
+of commands that violate this principle: `run`, `expose`, `autoscale`
+
+#### Specific Types of Functionality
+
+SIG CLI develops tools within the Kubernetes project including but not limited
+to the following:
+
+- Invoking Kubernetes APIs:
+  - Resource APIs - e.g. `create`, `replace`, `delete`, `patch`, `get`, etc
+  - SubResource APIs - e.g. `exec`, `attach`, `logs`, `scale`, etc
+  - Discovery Service - e.g. `api-versions`, `api-resources
+  - Version - e.g. `version`
+  - OpenAPI - e.g. `explain`
+- Pre and Post processing API Resource Config and API invocation responses
+  - `apply` - creates a patch from Resource Config by performing a 3-way diff
+  - `get` - formats API responses into columns
+- Aggregating multiple API invocations and post processing the results
+  - `describe` - performs multiple calls and aggregates results
+  - `run --expose` - creates multiple resources
+- Collapsing multiple manual steps into a command
+  - `edit` - collapses `get`, edit a file, `patch` into a single command
+  - `cp` - collapses manual steps for copy a file out of a container
+- Generating Kubernetes Resource Config locally or creating Resources remotely
+  - From commands + arguments + flags - e.g. `create configmap`, `run`, `expose`
+  - From declarative files - Kustomization `configMapGenerator`, `secretMapGenerator`
+- Transforming Kubernetes Resource Config locally or patching remotely
+  - From commands + arguments + flags - e.g. `annotate`, `set image`, `patch`
+  - From declarative files - e.g. Kustomization `commonAnnotations`, `images`, `patches`
+  - From flags - e.g. `-n` - set the namespace of Resources
+- Blocking on propagation of an event or change to the cluster
+  - e.g. `wait`, `rollout`
+- Defining or referencing a collection of Resource Config
+  - e.g. `-f` can reference a url, a file containing multiple Resources, or a directory
+  - e.g. `-R` can traverse a directory recursively
+  - e.g. Kustomization `bases` can refer to other Kustomizations
+  - e.g. Kustomization `resources` can refer to Resource Config files
+- Configure how to talk to a specific cluster from the cli
+  - e.g. `config`
+  - e.g. `--contect`
+- Selecting which API group/version to invoke if ambiguous in the context of the command
+  - e.g. `get`, `describe`
+  
+Notably, CLI Tooling may perform operations (generation, transformation, validation)
+that are augmented by analyzing collections of Resource Config - e.g. updating
+things that reference names of resources if transforming the names of resources.
+
+#### Go Libraries for developing CLIs
+
+SIG CLI develops go libraries for developing CLI tools for working with Kubernetes.
+These libraries provide a subset of the libraries used to build the CLI tools
+themselves.
```

Comment:
Maybe worth adding that kubectl will be always the first consumer of those.

Reply:
kubectl or cli tools in general?